azkaban-common/src/main/java/azkaban/executor/ExecutionControllerUtils.java

```diff
@@ -136,6 +136,40 @@ public static void alertUser(final ExecutableFlow flow, final AlerterHolder aler
     }
   }
 
+  /**
+   * Alert the user when the flow has encountered the first error.
+   *
+   * @param flow the execution
+   * @param alerterHolder the alerter holder
+   */
+  public static void alertUserOnFirstError(final ExecutableFlow flow,
```

Comment:
This method is moved from RunningExecutionsUpdater.java with slight modification.